system/include/emscripten/bind.h

```diff
@@ -386,6 +419,36 @@ struct Invoker<void, Args...> {
     }
 };
 
+namespace async {
+
+template<typename ReturnType, typename... Args>
+struct Invoker {
```

Comment:
I feel like there must be a better way to do this than duplicating all the *Invoker templates. If anyone has ideas let me know.

Reply:
Found a way to create a generic wrapper.